### ISSUE TITLE
Runtime Confusion ... to be or not to be !!!

### ISSUE THREAD

**Issue by aiamuzz** [flatpak/flatpak#2072]
## Linux distribution and version
Deepin OS

## Description of the problem
I already have runtime `org.gnome.Platform/x86_64/3.28` installed in my custom location for applications ... the app
`nl.hjdskes.gcolor3/x86_64/stable                   system,current`

installed at the default location(not at my custom location) runs just fine ... i guess by using the runtime from my custom location ...

However when i run `flatpak update` command ... i get the following ...

```
Looking for updates...
Required runtime for nl.hjdskes.gcolor3/x86_64/stable (org.gnome.Platform/x86_64/3.28) is not installed, searching...
Found in remote flathub, do you want to install it? [y/n]:
```
This shouldn't be happening isn't it ?
When the app can run fine with the runtime already present at another location ... it shouldn't be looking for it when the update command was issued ... **a scenario i thought you'd want to optimize and make it more efficient**

**Comment by alexlarsson** [comment]
Yeah, this should not be happening. 0.11.7 is kinda old though, need re-testing on 1.0.

**Comment by aiamuzz** [comment]
> Yeah, this should not be happening. 0.11.7 is kinda old though, need re-testing on 1.0.

I would love to test it ... but i am kind of stuck with this distro distributed primitive version ...

If somebody could help be resolve these issues ... may be then i could install the latest versions and test them ...

[Help Building on Deepin](https://github.com/flatpak/flatpak/issues/2068) ...
[Trading the Prepackaged Distro version with the latest from here ... Possible ???](https://github.com/flatpak/flatpak/issues/2049)

**Comment by Israphel** [comment]
But Deepin is debian unstable isn't it? maybe the debian package works

**Comment by aiamuzz** [comment]
> But Deepin is debian unstable isn't it? maybe the debian package works

@Israphel  ... yes its based on debian unstable ...

Surprisingly i was able to dig out the following packages as deb installers ...

https://packages.debian.org/sid/libostree-1-1
https://packages.debian.org/sid/flatpak

installation went without any dependency issues ... and ...

viola ... `flatpak --version` throws up as `Flatpak 1.0.1` ... seems with a straight forward solution i was going round and round for nothing ...
